### PR TITLE
Make it build with gch 8.8 and add ghc 8.8 travis build

### LINKED ISSUE #317: Support for GHC-8.8
As far as I can tell, there are two things blocking `hedgehog`'s compatibility with GHC-8.8.1:

* The bound on `template-haskell` needs to be lifted

* Happy needs updates to its `Setup.hs`, which it appears to be getting in https://github.com/simonmar/happy/pull/145.

### PR DESCRIPTION
**Pull request by erikd**
---

Edit (@moodmosaic): Resolves #317

### CODE REVIEW

File: hedgehog/src/Hedgehog/Internal/Gen.hs [hunk 1]
```diff
@@ -486,8 +487,10 @@ instance Monad m => Monad (GenT m) where
           runGenT size sk . k =<<
           runGenT size sm m
 
+#if MIN_VERSION_base(4,14,0)
```
Comment (moodmosaic) [inline hedgehog/src/Hedgehog/Internal/Gen.hs]: I thought it should be `4,13,0` https://gitlab.haskell.org/ghc/ghc/wikis/commentary/libraries/version-history

Comment (erikd) [reply hedgehog/src/Hedgehog/Internal/Gen.hs]: That compiled with both ghc 8.6 and 8.8 but may not actually have been correct.

Think I have fixed it now.

**Comment by newhoggy** [comment]
Please add minbounds on `checkers` because earlier versions don't build on `ghc-8.8`

**Comment by erikd** [comment]
@newhoggy Sure, what should the minbound be?

**Comment by newhoggy** [comment]
`checkers-0.5.2` please.  It's the latest version.  `checkers-0.5.1` was the first version to build on `ghc-8.8` but it broke `ghc-7.10` I think.

**Comment by erikd** [comment]
But @newhoggy , [hedgehog](https://hackage.haskell.org/package/hedgehog) doesn't depend on checkers.

**Comment by newhoggy** [comment]
Probably not in `hedgehog` proper, but this line.

```
$ grep checkers */*.cabal
hedgehog-test-laws/hedgehog-test-laws.cabal:    , checkers                        >= 0.5        && < 0.6
```

**Comment by erikd** [comment]
Ah ok, fixed!

**Comment by newhoggy** [comment]
Thanks!

**Comment by newhoggy** [comment]
👍

**Comment by newhoggy** [comment]
How did you edit the metadata to avoid solving the bad case?

**Comment by gwils** [comment]
I revised checkers 0.5.1 to have a base lower bound of >=4.9, so it will no
longer try to build on GHC 7.10

**Comment by erikd** [comment]
@newhoggy George used his super powers as a Hackage Trustee to fix it!

**Comment by moodmosaic** [comment]
@newhoggy, if you can put `checkers` back then fine, otherwise I think it's OK to leave it as-is since it's only used in `hedgehog-test-laws/hedgehog-test-laws.cabal`. @gwils @jacobstanley

**Comment by newhoggy** [comment]
@erikd I think you need to do it since it's your PR

**Comment by erikd** [comment]
Any plans for a new Hackage release. I can't release ghc 8.8 versions of some of my packages until this is released to Hackage.

**Comment by gwils** [comment]
^ This applies to pretty much all qfpl packages as well.

**Comment by erikd** [comment]
I deleted my comment :). Version 1.0.1 has been released.

**Comment by moodmosaic** [comment]
>I deleted my comment :). Version 1.0.1 has been released.

@erikd, yes. So when a *release* PR like https://github.com/hedgehogqa/haskell-hedgehog/pull/324 is merged, it means there's a new version on Hackage plus an annotated tag for it here on GitHub.

**Comment by erikd** [comment]
Yeah, I did not see that PR until just now :).

**Pull request merged**